lib/SILOptimizer/Analysis/MemoryBehavior.cpp

```diff
@@ -241,10 +241,17 @@ MemBehavior MemoryBehaviorVisitor::visitApplyInst(ApplyInst *AI) {
   MemBehavior Behavior = MemBehavior::None;
 
   // We can ignore mayTrap().
-  if (ApplyEffects.mayReadRC() ||
-      (InspectionMode == RetainObserveKind::ObserveRetains &&
-       ApplyEffects.mayAllocObjects())) {
-    Behavior = MemBehavior::MayHaveSideEffects;
+  if (any_of(enumerate(AI->getArgumentOperands()),
```

Comment:
Sorry, I didn't realize that this is not the right way to get the parameter convention. The SILFunctionType::getParameters() don't take the indirect results into account.
The correct way is to the the SILFunctionConventions with getSubstCalleeConv().
Then use getSILArgumentConvention(op.Index)